lib/testable-test.js

```diff
@@ -0,0 +1,258 @@
+      suiteUuid: suite.uuid,
+      suiteName: suite.name,
+      uuid: uuidv4(),
+      name: name,
+      fn: fn,
+      started: Date.now()
+    };
+  }
+
+  async _executeSuite(suite) {
+    await this._onSuiteStart(suite);
+
+    if ( suite.tests )
+      for ( const test of suite.tests ) {
+        await this._executeTest(suite, test);
+      }
+
+    await this._onSuiteFinish(suite);
+  }
+
+  async _onSuiteStart(suite) {
+    const before = suite._before || this._before;
+    if ( before )
+      await this._executeFn(before);
+
+    if ( this.cb.startSuite )
+      this.cb.startSuite(suite);
+  }
+
+  async _onSuiteFinish(suite) {
+    suite.finished = Date.now();
+    suite.duration = Date.now() - suite.started;
+
+    const after = suite._after || this._after;
+    if ( after )
+      await this._executeFn(after);
+
+    this.suites.pop();
+    if ( this.cb.finishSuite )
+      this.cb.finishSuite(suite);
+  }
+
+  async _executeTest(suite, test) {
+    const beforeEach = suite._beforeEach || this._beforeEach;
+    if ( beforeEach )
+      await this._executeFn(beforeEach);
+
+    await this._executeStep(test.fn, suite, test);
+
+    const afterEach = suite._afterEach || this._afterEach;
+    if ( afterEach )
+      await this._executeFn(afterEach);
+  }
+
+  async _executeStep(fn, suite, test) {
+    if ( this.cb.startTest )
+      this.cb.startTest(test, suite);
+
+    try {
+      await this._executeFn(fn);
+      test.state = 'passed';
+    } catch (err) {
+      suite.hasError = true;
+      test.state = 'failed';
+      test.errorType = err.name;
+      if ( err.message )
+        test.error = err.message;
+      if ( err.stack )
+        test.errorTrace = this.toErrorContent(err);
+    }
+    test.finished = Date.now();
+    test.duration = Date.now() - test.started;
+
+    if ( this.cb.finishTest )
+      this.cb.finishTest(test, suite);
+  }
+
+  async _executeFn(fn) {
+    const isAsync = fn.constructor.name === 'AsyncFunction';
+    if ( isAsync ) {
+      await fn();
+    } else if ( fn.length === 1 ) {
+      await new Promise((resolve, reject) => {
+        fn((err) => {
+          if ( err )
+            reject(new Error(err));
+          else
+            resolve();
+        });
+      });
+    } else {
+      fn();
+    }
+  }
+
+  reportTestSuiteActions(type, contents) {
+    const currentSuite = this.currentSuite;
+    if ( currentSuite && type === 'FinishSuiteTest' && contents && contents.state === 'failed' )
+      currentSuite.hasError = true;
+
+    const result = { type: type, data: contents };
+    return new Promise((resolve, reject) => {
+      if ( process.env.TESTABLE_RESULT_FILE ) {
+        fs.writeFile(process.env.TESTABLE_RESULT_FILE, JSON.stringify(result) + '\n', {
+          encoding: 'utf8',
+          flag: 'a'
+        }, resolve);
+      } else {
+        console.log('[Result] ' + JSON.stringify(result, null, '\t'));
+        resolve();
+      }
+    });
+  }
+
+  toErrorContent(error) {
```

Comment:
All these `if` statements are not applicable here. We know given the single location that calls this above that `error` will be an `Error` object. The first 5 if statements will never get called.